XamarinCommunityToolkit/Behaviors/MaxLengthReachedBehavior.shared.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+    public class MaxLengthReachedBehavior : BaseBehavior<InputView>
+	{
+		public static readonly BindableProperty CommandProperty
+			= BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(UserStoppedTypingBehavior));
+
+		public static readonly BindableProperty ShouldDismissKeyboardAutomaticallyProperty
+			= BindableProperty.Create(nameof(ShouldDismissKeyboardAutomatically), typeof(bool), typeof(UserStoppedTypingBehavior), false);
+
+		public event EventHandler<MaxLengthReachedEventArgs> MaxLengthReached;
```

Comment:
Should we use a weak event manager here?

Reply:
Hi @AndreiMisiukevich, could you elaborate a bit more on this? What the WeakEventManager exactly does and how to implement this here?

Reply:
@jBijsterboschNL [Long story](https://www.codeproject.com/Articles/738109/The-NET-Weak-Event-Pattern-in-Csharp) short with the WeakEventManager the GC can collect the `EventHandler` reference and we don't need to worry about memory leak.

The XCT has an implementation for it, you see the implementation [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkit/Helpers/WeakEventManager.shared.cs) and a sample of use [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkit/Views/Expander/Expander.shared.cs#L16)

Reply:
Thanks @pictos. I've implemented the WeakEventManager on the MaxLengthReached EventHandler